app/models/user.rb

```diff
@@ -0,0 +1,18 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
```

Comment:
như bạn đằng trc, giải thích cái regex này đi bạn 👿

Reply:
đấy là điều kiện email ạ, ví dụ \w là bất cứ kí tự nào, \A[\w+\-.] là để cho phép mình dùng kí tự đặc biệt như "-" và "." còn [a-z\d\-] là chứa các kí tự từ a đến z, các số từ 0 đến 9 và dấu "-". đằng sau cũng tương tự ạ. Em hiểu đây là điều kiện bắt buôcj để mình tuân theo ạ

Reply:
dấu + cũng là kí tự đặc biệt nhé không phải là chỉ số lượng đâu =))

Reply:
các chú toàn hiểu nhầm dấu + trong []

Reply:
\A và \z là thể hiện bắt đầu kết thúc chuỗi nhé, đừng nhét vào block khi giải thích

Reply:
vâng em hiểu rồi ạ, em cảm ơn a